src/cargo/ops/cargo_doc.rs

```diff
@@ -59,7 +59,7 @@ pub fn doc(ws: &Workspace<'_>, options: &DocOptions) -> CargoResult<()> {
         let name = &compilation
             .root_crate_names
             .get(0)
-            .ok_or_else(|| anyhow::anyhow!("no crates with documentation"))?;
+            .ok_or_else(|| anyhow::anyhow!("requested crate documentation is not available to open"))?;
```

Comment:
The hard part of this is fix is in the nuance.  By saying its "not available" implies it is there but something prevents it.

What about
```suggestion
            .ok_or_else(|| anyhow::anyhow!("cannot open specified crate's documentation: no documentation generated"))?;
```

Reply:
What if the examples file is not even present? How it can open? Will this message still be suitable?

Reply:
To fix the problem, it needs to discuss opening as that is the operation that is failing.

Reply:
Alright, Now I understand the point.